### ISSUE TITLE
multiple node training slower than single node

### ISSUE THREAD

**Issue by zhangyan-ucas** [modelscope/ms-swift#4291]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)

Multiple node training slower than single node. Below is training scripts and log for multiple nodes and single node. From the training log, we find that multiple node training extremely slower than single node. Any help would be greatly appreciated!

multiple node training script: 
run_name="v6_sft_ac"
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NNODES=2 \
NODE_RANK=${RANK} \
MASTER_ADDR=${MASTER_ADDR} \
MASTER_PORT=${MASTER_PORT} \
NPROC_PER_NODE=8 \
MAX_PIXELS=2371600 \
swift sft \
    --model Qwen2.5-VL-7B-Instruct-new \
    --train_type full \
    --dataset "train_e1_p4.jsonl" \
    --torch_dtype bfloat16 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-5 \
    --gradient_accumulation_steps 2 \
    --packing false \
    --streaming false \
    --lazy_tokenize true \
    --eval_strategy no \
    --eval_steps 500 \
    --save_strategy "no" \
    --save_total_limit 3 \
    --logging_steps 5 \
    --deepspeed zero2 \
    --max_length 16384 \
    --num_train_epochs 3.0 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 32 \
    --dataset_num_proc 32 \
    --save_only_model true \
    --output_dir output/${run_name} \
    --run_name ${run_name} \
    --report_to none \
    --attn_impl flash_attn \
    --freeze_llm false \
    --freeze_vit true \
    --freeze_aligner true

multiple node training log: 
2025-05-21T02:08:42.562055091Z {'loss': 2.34024644, 'token_acc': 0.61165049, 'grad_norm': 52.97337341, 'learning_rate': 9e-08, 'memory(GiB)': 38.37, 'train_speed(iter/s)': 0.008331, 'epoch': 0.0, 'global_step/max_steps': '1/2244', 'percentage': '0.04%', 'elapsed_time': '1m 24s', 'remaining_time': '2d 4h 35m 14s'}
2025-05-21T02:10:40.022526108Z {'loss': 2.39078999, 'token_acc': 0.67772512, 'grad_norm': 52.0651741, 'learning_rate': 4.4e-07, 'memory(GiB)': 75.46, 'train_speed(iter/s)': 0.021045, 'epoch': 0.01, 'global_step/max_steps': '5/2244', 'percentage': '0.22%', 'elapsed_time': '3m 21s', 'remaining_time': '1d 1h 6m 38s'}
2025-05-21T02:13:03.225634822Z {'loss': 2.23430481, 'token_acc': 0.69488189, 'grad_norm': 47.41981888, 'learning_rate': 8.8e-07, 'memory(GiB)': 75.46, 'train_speed(iter/s)': 0.026261, 'epoch': 0.01, 'global_step/max_steps': '10/2244', 'percentage': '0.45%', 'elapsed_time': '5m 45s', 'remaining_time': '21h 24m 49s'}
2025-05-21T02:15:25.999893984Z {'loss': 1.59879217, 'token_acc': 0.71755725, 'grad_norm': 24.92251587, 'learning_rate': 1.33e-06, 'memory(GiB)': 75.46, 'train_speed(iter/s)': 0.02865, 'epoch': 0.02, 'global_step/max_steps': '15/2244', 'percentage': '0.67%', 'elapsed_time': '8m 7s', 'remaining_time': '20h 8m 13s'}
2025-05-21T02:17:48.806381408Z {'loss': 0.9737011, 'token_acc': 0.71602434, 'grad_norm': 15.97769165, 'learning_rate': 1.77e-06, 'memory(GiB)': 75.46, 'train_speed(iter/s)': 0.030013, 'epoch': 0.03, 'global_step/max_steps': '20/2244', 'percentage': '0.89%', 'elapsed_time': '10m 30s', 'remaining_time': '19h 28m 48s'}
2025-05-21T02:20:09.768629630Z {'loss': 0.57388086, 'token_acc': 0.82191781, 'grad_norm': 4.69608116, 'learning_rate': 2.21e-06, 'memory(GiB)': 75.46, 'train_speed(iter/s)': 0.030966, 'epoch': 0.03, 'global_step/max_steps': '25/2244', 'percentage': '1.11%', 'elapsed_time': '12m 51s', 'remaining_time': '19h 1m 28s'}
2025-05-21T02:22:31.072132999Z {'loss': 0.49524522, 'token_acc': 0.83387622, 'grad_norm': 4.18681717, 'learning_rate': 2.65e-06, 'memory(GiB)': 75.46, 'train_speed(iter/s)': 0.031624, 'epoch': 0.04, 'global_step/max_steps': '30/2244', 'percentage': '1.34%', 'elapsed_time': '15m 12s', 'remaining_time': '18h 42m 53s'}

single node training script: 
run_name="v7_sft_ac"
NPROC_PER_NODE=8 \
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
MAX_PIXELS=2371600 \
swift sft \
    --model hf/Qwen2.5-VL-7B-Instruct \
    --train_type full \
    --dataset "train_e1_p4.jsonl" \
    --torch_dtype bfloat16 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-5 \
    --gradient_accumulation_steps 2 \
    --packing false \
    --streaming false \
    --lazy_tokenize true \
    --eval_strategy no \
    --eval_steps 500 \
    --save_strategy "no" \
    --save_total_limit 3 \
    --logging_steps 5 \
    --deepspeed zero3 \
    --max_length 16384 \
    --num_train_epochs 3.0 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 32 \
    --dataset_num_proc 32 \
    --save_only_model true \
    --output_dir output/${run_name} \
    --run_name ${run_name} \
    --report_to none \
    --attn_impl flash_attn \
    --freeze_llm false \
    --freeze_vit true \
    --freeze_aligner true 

Single node training log: 
2025-05-21T03:50:19.029644300Z {'loss': 0.42409496, 'token_acc': 0.85283019, 'grad_norm': 3.6537035, 'learning_rate': 9.99e-06, 'memory(GiB)': 76.72, 'train_speed(iter/s)': 0.193983, 'epoch': 0.2, 'global_step/max_steps': '295/4491', 'percentage': '6.57%', 'elapsed_time': '25m 15s', 'remaining_time': '5h 59m 14s'}
2025-05-21T03:50:42.015709928Z {'loss': 0.42228584, 'token_acc': 0.87710843, 'grad_norm': 5.19625193, 'learning_rate': 9.99e-06, 'memory(GiB)': 76.72, 'train_speed(iter/s)': 0.194334, 'epoch': 0.2, 'global_step/max_steps': '300/4491', 'percentage': '6.68%', 'elapsed_time': '25m 38s', 'remaining_time': '5h 58m 11s'}
2025-05-21T03:51:06.122969311Z {'loss': 0.371628, 'token_acc': 0.85766423, 'grad_norm': 4.15334296, 'learning_rate': 9.99e-06, 'memory(GiB)': 76.72, 'train_speed(iter/s)': 0.194535, 'epoch': 0.2, 'global_step/max_steps': '305/4491', 'percentage': '6.79%', 'elapsed_time': '26m 2s', 'remaining_time': '5h 57m 24s'}
2025-05-21T03:51:30.185426376Z {'loss': 0.35028491, 'token_acc': 0.87245841, 'grad_norm': 2.94855862, 'learning_rate': 9.99e-06, 'memory(GiB)': 76.72, 'train_speed(iter/s)': 0.194736, 'epoch': 0.21, 'global_step/max_steps': '310/4491', 'percentage': '6.90%', 'elapsed_time': '26m 26s', 'remaining_time': '5h 56m 37s'}
2025-05-21T03:51:54.304365337Z {'loss': 0.3620203, 'token_acc': 0.88686869, 'grad_norm': 3.7277799, 'learning_rate': 9.99e-06, 'memory(GiB)': 76.72, 'train_speed(iter/s)': 0.194923, 'epoch': 0.21, 'global_step/max_steps': '315/4491', 'percentage': '7.01%', 'elapsed_time': '26m 50s', 'remaining_time': '5h 55m 52s'}
2025-05-21T03:52:18.569766412Z {'loss': 0.37554173, 'token_acc': 0.85448916, 'grad_norm': 3.17496678, 'learning_rate': 9.99e-06, 'memory(GiB)': 76.72, 'train_speed(iter/s)': 0.195088, 'epoch': 0.21, 'global_step/max_steps': '320/4491', 'percentage': '7.13%', 'elapsed_time': '27m 14s', 'remaining_time': '5h 55m 10s'}

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)

CUDA: 12.2

Here is my environment:
absl-py                       2.2.2
accelerate                    1.6.0
addict                        2.4.0
aiofiles                      24.1.0
aiohappyeyeballs              2.6.1
aiohttp                       3.11.18
aiosignal                     1.3.2
aliyun-python-sdk-core        2.16.0
aliyun-python-sdk-kms         2.16.5
annotated-types               0.7.0
anyio                         4.9.0
asttokens                     3.0.0
async-timeout                 5.0.1
attrdict                      2.0.1
attrs                         25.3.0
audioread                     3.0.1
av                            14.3.0
binpacking                    1.5.2
blessed                       1.21.0
certifi                       2025.4.26
cffi                          1.17.1
charset-normalizer            3.4.2
click                         8.1.8
colorama                      0.4.6
contourpy                     1.3.2
cpm-kernels                   1.0.11
crcmod                        1.7
cryptography                  44.0.3
cycler                        0.12.1
dacite                        1.9.2
datasets                      3.3.2
decorator                     5.2.1
decord                        0.6.0
deepspeed                     0.16.7
dill                          0.3.8
distro                        1.9.0
docker-pycreds                0.4.0
einops                        0.8.1
exceptiongroup                1.2.2
executing                     2.2.0
fastapi                       0.115.12
ffmpy                         0.5.0
filelock                      3.18.0
flash_attn                    2.7.4.post1
fonttools                     4.57.0
frozenlist                    1.6.0
fsspec                        2024.12.0
future                        1.0.0
gitdb                         4.0.12
GitPython                     3.1.44
gpustat                       1.1.1
gradio                        5.29.0
gradio_client                 1.10.0
groovy                        0.1.2
grpcio                        1.71.0
h11                           0.16.0
hf-xet                        1.1.0
hjson                         3.1.0
httpcore                      1.0.9
httpx                         0.28.1
huggingface-hub               0.31.1
icecream                      2.1.4
idna                          3.10
importlib_metadata            8.7.0
jieba                         0.42.1
Jinja2                        3.1.6
jiter                         0.9.0
jmespath                      0.10.0
joblib                        1.5.0
kiwisolver                    1.4.8
lazy_loader                   0.4
librosa                       0.11.0
llvmlite                      0.44.0
Markdown                      3.8
markdown-it-py                3.0.0
MarkupSafe                    3.0.2
matplotlib                    3.10.1
mdurl                         0.1.2
modelscope                    1.25.0
mpmath                        1.3.0
ms-swift                      3.5.0.dev0  /mnt/vlm-ks3/zhangyan/open/ms-swift
msgpack                       1.1.0
multidict                     6.4.3
multiprocess                  0.70.16
networkx                      3.4.2
ninja                         1.11.1.4
nltk                          3.9.1
numba                         0.61.2
numpy                         1.26.4
nvidia-cublas-cu12            12.6.4.1
nvidia-cuda-cupti-cu12        12.6.80
nvidia-cuda-nvrtc-cu12        12.6.77
nvidia-cuda-runtime-cu12      12.6.77
nvidia-cudnn-cu12             9.5.1.17
nvidia-cufft-cu12             11.3.0.4
nvidia-cufile-cu12            1.11.1.6
nvidia-curand-cu12            10.3.7.77
nvidia-cusolver-cu12          11.7.1.2
nvidia-cusparse-cu12          12.5.4.2
nvidia-cusparselt-cu12        0.6.3
nvidia-ml-py                  12.575.51
nvidia-nccl-cu12              2.26.2
nvidia-nvjitlink-cu12         12.6.85
nvidia-nvtx-cu12              12.6.77
openai                        1.77.0
orjson                        3.10.18
oss2                          2.19.1
packaging                     25.0
pandas                        2.2.3
peft                          0.15.2
pillow                        11.2.1
pip                           23.0.1
platformdirs                  4.3.8
pooch                         1.8.2
propcache                     0.3.1
protobuf                      6.30.2
psutil                        7.0.0
py-cpuinfo                    9.0.0
pyarrow                       20.0.0
pyav                          14.2.1
pycparser                     2.22
pycryptodome                  3.22.0
pydantic                      2.11.4
pydantic_core                 2.33.2
pydub                         0.25.1
Pygments                      2.19.1
pyparsing                     3.2.3
python-dateutil               2.9.0.post0
python-multipart              0.0.20
pytz                          2025.2
PyYAML                        6.0.2
qwen-vl-utils                 0.0.11
regex                         2024.11.6
requests                      2.32.3
rich                          14.0.0
rouge                         1.0.1
ruff                          0.11.8
safehttpx                     0.1.6
safetensors                   0.5.3
scikit-learn                  1.6.1
scipy                         1.15.2
semantic-version              2.10.0
sentencepiece                 0.2.0
sentry-sdk                    2.28.0
setproctitle                  1.3.6
setuptools                    65.5.0
shellingham                   1.5.4
simplejson                    3.20.1
six                           1.17.0
smmap                         5.0.2
sniffio                       1.3.1
sortedcontainers              2.4.0
soundfile                     0.13.1
soxr                          0.5.0.post1
starlette                     0.46.2
sympy                         1.14.0
tensorboard                   2.19.0
tensorboard-data-server       0.7.2
threadpoolctl                 3.6.0
tiktoken                      0.9.0
tokenizers                    0.21.1
tomlkit                       0.13.2
torch                         2.7.0
torchvision                   0.22.0
tqdm                          4.67.1
transformers                  4.51.3
transformers-stream-generator 0.0.5
triton                        3.3.0
trl                           0.17.0
typer                         0.15.3
typing_extensions             4.13.2
typing-inspection             0.4.0
tzdata                        2025.2
urllib3                       2.4.0
uvicorn                       0.34.2
wandb                         0.19.11
wcwidth                       0.2.13
websockets                    15.0.1
Werkzeug                      3.1.3
wheel                         0.45.1
xxhash                        3.5.0
yarl                          1.20.0
zipp                          3.21.0
zstandard                     0.23.0

**Additional context**
Add any other context about the problem here(在这里补充其他信息)
Any help would be greatly appreciated!
